Source/Core/AbsyExpr.cs

```diff
@@ -3837,6 +3843,197 @@ public T Dispatch<T>(IAppliableVisitor<T> visitor)
     }
   }
 
+  public class FieldAccess : IAppliable
+  {
+    private IToken _tok;
+
+    private string _fieldName;
+    
+    private DatatypeSelector _selector;
+
+    public IToken tok
+    {
+      get
+      {
+        Contract.Ensures(Contract.Result<IToken>() != null);
+        return this._tok;
+      }
+      set
+      {
+        Contract.Requires(value != null);
+        this._tok = value;
+      }
+    }
+    
+    public string fieldName
+    {
+      get
+      {
+        Contract.Ensures(Contract.Result<IToken>() != null);
+        return this._fieldName;
+      }
+      set
+      {
+        Contract.Requires(value != null);
+        this._fieldName = value;
+      }
+    }
+    
+    public DatatypeSelector selector
+    {
+      get
+      {
+        Contract.Ensures(Contract.Result<IToken>() != null);
+        return this._selector;
+      }
+      set
+      {
+        Contract.Requires(value != null);
+        this._selector = value;
+      }
+    }
+
+    [ContractInvariantMethod]
+    void ObjectInvariant()
+    {
+      Contract.Invariant(this._tok != null);
+    }
+
+    public FieldAccess(IToken tok, string fieldName)
+    {
+      Contract.Requires(tok != null);
+      this._tok = tok;
+      this._fieldName = fieldName;
+    }
+    
+    public FieldAccess(IToken tok, DatatypeSelector selector)
+    {
+      Contract.Requires(tok != null);
+      this._tok = tok;
+      this._selector = selector;
+      this._fieldName = selector.Name;
+    }
+
+    public string FunctionName
+    {
+      get
+      {
+        Contract.Ensures(Contract.Result<string>() != null);
+
+        return "field-access";
+      }
+    }
+
+    [Pure]
+    [Reads(ReadsAttribute.Reads.Nothing)]
+    public override bool Equals(object obj)
+    {
+      if (obj is FieldAccess fieldAccess)
+      {
+        return selector.Equals(fieldAccess.selector);
+      }
+      return false;
+    }
+
+    [Pure]
+    public override int GetHashCode()
+    {
+      return 1;
+    }
+
+    public void Emit(IList<Expr> args, TokenTextWriter stream, int contextBindingStrength, bool fragileContext)
+    {
+      stream.SetToken(this);
+      Contract.Assert(args.Count == 1);
+      stream.push();
+      stream.Write("(");
```

Comment:
The `contextBindingStrength` parameter can be used to let you omit these parentheses. I think something like that is done with the `...[...]` expression.

Reply:
Done. I am using the same opBinding strength for `->` as for `...[...]`. 

```
const int opBindingStrength = 0x90;
```

Sound good?

Reply:
Yes, same binding strength as `...[...]` is good. But note my comment in the code--the recursive call should pass in the `opBindingStrength` you use, not `0x00`.

Reply:
Done.